### DISCUSSION TITLE
How to setup parameter for batch endpoint

### DISCUSSION THREAD

**Discussion by mail2mhossain** [langchain-ai/langserve#546] [Q&A] [answered]
import requests

inputs = {"input": {"age": age, "json_example": json_example_4_subject}}

response = requests.post("http://localhost:8000/generate_subjects/batch", json=inputs)
#response = requests.post("http://localhost:8000/generate_subjects/invoke", json=inputs)

print(response.json())

In the above code, "invoke" is working BUT "batch" is not working.

**Comment by eyurtsev** [comment]
inputs needs to be a list 

inputs = [{"input": {"age": age, "json_example": json_example_4_subject}}]

**Comment by mail2mhossain** [comment]
inputs = [{"input": {"age": age, "json_example": json_example_4_subject}}]
Getting <Response [500]> in client side.

In server side:
pydantic.v1.errors.DictError: value is not a valid dict

**Comment by pietromaldiniMoviri** [comment, marked as answer]
I think the correct way should be.

```
import requests
single_input = {"input": {"age": age, "json_example": json_example_4_subject}}
batch_input = {"inputs": [{"age": age, "json_example": json_example_4_subject},{"age": age2, "json_example": json_example_5_subject}]}


response_single = requests.post("http://localhost:8000/generate_subjects/invoke", json=single_input)
response_batch = requests.post("http://localhost:8000/generate_subjects/batch", json=batch_input)

print(response_single.json())
print(response_batch.json())
```

the batch endpoint expects an object with a key "inputs" that has as a value a list of objects with the type the invoke endpoint expects.
